src/lightning/pytorch/CHANGELOG.md

```diff
@@ -14,6 +14,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Added robust timer duration parsing with an informative error message when parsing fails ([#19513](https://github.com/Lightning-AI/pytorch-lightning/pull/19513))
 
+- Added `on_exception` hook to `LightningDataModule` to gracefully handle the exception that interrupted the training
```

Comment:
I would drop the explanation here. The hooks are not actually meant to handle the exception itself, but to run additional code before raising the exception. 
```suggestion
- Added `on_exception` hook to `LightningDataModule`
```

Reply:
Also let's add the link to your PR:

```suggestion
- Added `on_exception` hook to `LightningDataModule` ([#19601](https://github.com/Lightning-AI/pytorch-lightning/pull/19601))
```